File: app/src/main/java/com/pennywiseai/tracker/ui/screens/settings/FAQScreen.kt
package com.pennywiseai.tracker.ui.screens.settings

import android.content.Intent
import android.net.Uri
import androidx.compose.animation.*
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import com.pennywiseai.tracker.ui.effects.overScrollVertical
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.automirrored.filled.Message
import androidx.compose.material.icons.automirrored.filled.OpenInNew
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.input.nestedscroll.nestedScroll
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.pennywiseai.tracker.ui.components.CustomTitleTopAppBar
import com.pennywiseai.tracker.ui.components.cards.PennyWiseCardV2
import com.pennywiseai.tracker.ui.components.cards.SectionHeaderV2
import com.pennywiseai.tracker.ui.theme.Dimensions
import com.pennywiseai.tracker.ui.theme.Spacing
import dev.chrisbanes.haze.HazeState
import dev.chrisbanes.haze.hazeSource

data class FAQItem(
    val question: String,
    val answer: String
)

data class FAQCategory(
    val title: String,
    val icon: @Composable () -> Unit,
    val items: List<FAQItem>
)

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun FAQScreen(
    onNavigateBack: () -> Unit,
    modifier: Modifier = Modifier
) {
    val context = LocalContext.current
    
    val faqCategories = remember {
        listOf(
            FAQCategory(
                title = "Transaction Types",
                icon = { Icon(Icons.Default.SwapHoriz, contentDescription = null) },
                items = listOf(
                    FAQItem(
                        question = "Why are wallet transactions marked as Credit?",
                        answer = "Wallet transactions (Amazon Pay, Paytm, etc.) are marked as Credit because they're charged to your bank account or credit card first, not direct bank debits. This helps track the actual payment method used."
                    ),
                    FAQItem(
                        question = "What's the difference between the 5 transaction types?",
                        answer = """• Expense: Money going out of your account (debits, purchases, bill payments)
• Income: Money coming into your account (salary, refunds, cashback)
• Investment: Mutual funds, stocks, SIPs, trading accounts
• Credit: Credit card transactions and wallet payments (money you'll pay later)
• Transfer: Money moved between your own accounts (self-transfers)"""
                    ),
                    FAQItem(
                        question = "When should I use Transfer vs Expense?",
                        answer = "Use Transfer when moving money between your own accounts (e.g., savings to checking). These don't affect your net worth. Use Expense for actual spending."
                    )
                )
            ),
            FAQCategory(
                title = "SMS Parsing",
                icon = { Icon(Icons.AutoMirrored.Filled.Message, contentDescription = null) },
                items = listOf(
                    FAQItem(
                        question = "Why aren't my bank SMS being detected?",
                        answer = "Check if your bank is supported in our list. If not, report it via GitHub. Ensure SMS permissions are granted and the sender format matches standard bank SMS patterns."
                    ),
                    FAQItem(
                        question = "What happens to unrecognized SMS?",
                        answer = "They're saved in 'Unrecognized Messages' where you can manually review them or report them to help us improve parsing."
                    ),
                    FAQItem(
                        question = "Why are some transactions duplicated?",
                        answer = "Some banks send multiple SMS for the same transaction. The app tries to detect duplicates, but you can manually delete any that slip through."
                    )
                )
            ),
            FAQCategory(
                title = "Privacy & Data",
                icon = { Icon(Icons.Default.Security, contentDescription = null) },
                items = listOf(
                    FAQItem(
                        question = "Is my financial data secure?",
                        answer = "Yes! All data stays on your device. We don't have servers or cloud storage. The AI model runs locally for complete privacy."
                    ),
                    FAQItem(
                        question = "Can I backup my data?",
                        answer = "Currently, data is stored locally only. Export/backup features are planned for future updates."
                    ),
                    FAQItem(
                        question = "What data does the app access?",
                        answer = "Only SMS messages from known bank senders. We don't read personal messages or access other app data."
                    )
                )
            ),
            FAQCategory(
                title = "AI Features",
                icon = { Icon(Icons.Default.Psychology, contentDescription = null) },
                items = listOf(
                    FAQItem(
                        question = "Why do I need to download the AI model?",
                        answer = "The 750MB model enables on-device chat about your expenses without sending data to any server, ensuring complete privacy."
                    ),
                    FAQItem(
                        question = "What can I ask the AI assistant?",
                        answer = "You can ask about spending patterns, budget advice, transaction summaries, and general financial questions based on your data."
                    )
                )
            ),
            FAQCategory(
                title = "Account Management",
                icon = { Icon(Icons.Default.AccountBalance, contentDescription = null) },
                items = listOf(
                    FAQItem(
                        question = "What are manual accounts?",
                        answer = "Manual accounts let you track cash, investments, or accounts from unsupported banks. You update balances manually."
                    ),
                    FAQItem(
                        question = "How do I track multiple accounts from the same bank?",
                        answer = "The app automatically detects different accounts based on the last 4 digits shown in SMS."
                    )
                )
            )
        )
    }
    
    var expandedCategories by remember { mutableStateOf(setOf<Int>()) }

    val scrollBehaviorSmall = TopAppBarDefaults.pinnedScrollBehavior()
    val scrollBehaviorLarge = TopAppBarDefaults.exitUntilCollapsedScrollBehavior()
    val hazeState = remember { HazeState() }

    Scaffold(
        modifier = modifier.nestedScroll(scrollBehaviorLarge.nestedScrollConnection),
        containerColor = Color.Transparent,
        topBar = {
            CustomTitleTopAppBar(
                scrollBehaviorSmall = scrollBehaviorSmall,
                scrollBehaviorLarge = scrollBehaviorLarge,
                title = "FAQ",
                hasBackButton = true,
                hasActionButton = true,
                navigationContent = {
                    IconButton(onClick = onNavigateBack) {
                        Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "Back")
                    }
                },
                hazeState = hazeState
            )
        }
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .hazeSource(hazeState)
                .background(MaterialTheme.colorScheme.background)
                .overScrollVertical()
                .verticalScroll(rememberScrollState())
                .padding(paddingValues)
                .padding(Dimensions.Padding.content),
            verticalArrangement = Arrangement.spacedBy(Spacing.md)
        ) {
            // FAQ Categories
            faqCategories.forEachIndexed { categoryIndex, category ->
                SectionHeaderV2(title = category.title)
                
                PennyWiseCardV2(
                    modifier = Modifier.fillMaxWidth()
                ) {
                    Column {
                        category.items.forEachIndexed { itemIndex, faqItem ->
                            val isExpanded = expandedCategories.contains(categoryIndex * 100 + itemIndex)
                            
                            Column(
                                modifier = Modifier
                                    .fillMaxWidth()
                                    .clickable {
                                        expandedCategories = if (isExpanded) {
                                            expandedCategories - (categoryIndex * 100 + itemIndex)
                                        } else {
                                            expandedCategories + (categoryIndex * 100 + itemIndex)
                                        }
                                    }
                                    .padding(Dimensions.Padding.content)
                            ) {
                                Row(
                                    modifier = Modifier.fillMaxWidth(),
                                    horizontalArrangement = Arrangement.SpaceBetween,
                                    verticalAlignment = Alignment.Top
                                ) {
                                    Row(
                                        modifier = Modifier.weight(1f),
                                        horizontalArrangement = Arrangement.spacedBy(Spacing.md)
                                    ) {
                                        if (itemIndex == 0) {
                                            Box(
                                                modifier = Modifier.size(Dimensions.Icon.medium),
                                                contentAlignment = Alignment.Center
                                            ) {
                                                category.icon()
                                            }
                                        } else {
                                            Spacer(modifier = Modifier.width(Dimensions.Icon.medium))
                                        }
                                        
                                        Text(
                                            text = faqItem.question,
                                            style = MaterialTheme.typography.bodyLarge,
                                            fontWeight = FontWeight.Medium,
                                            color = MaterialTheme.colorScheme.onSurface
                                        )
                                    }
                                    
                                    Icon(
                                        imageVector = if (isExpanded) Icons.Default.ExpandLess else Icons.Default.ExpandMore,
                                        contentDescription = if (isExpanded) "Collapse" else "Expand",
                                        tint = MaterialTheme.colorScheme.onSurfaceVariant
                                    )
                                }
                                
                                AnimatedVisibility(
                                    visible = isExpanded,
                                    enter = expandVertically() + fadeIn(),
                                    exit = shrinkVertically() + fadeOut()
                                ) {
                                    Spacer(modifier = Modifier.height(Spacing.sm))
                                    Row(
                                        horizontalArrangement = Arrangement.spacedBy(Spacing.md)
                                    ) {
                                        Spacer(modifier = Modifier.width(24.dp))
                                        Text(
                                            text = faqItem.answer,
                                            style = MaterialTheme.typography.bodyMedium,
                                            color = MaterialTheme.colorScheme.onSurfaceVariant,
                                            modifier = Modifier.padding(top = Spacing.sm)
                                        )
                                    }
                                }
                            }
                            
                            if (itemIndex < category.items.size - 1) {
                                HorizontalDivider(
                                    modifier = Modifier.padding(horizontal = Dimensions.Padding.content)
                                )
                            }
                        }
                    }
                }
            }
            
            // Still need help section
            SectionHeaderV2(title = "Still Need Help?")
            
            PennyWiseCardV2(
                modifier = Modifier
                    .fillMaxWidth()
                    .clickable {
                        val intent = Intent(Intent.ACTION_VIEW, Uri.parse("https://github.com/sarim2000/pennywiseai-tracker/issues/new/choose"))
                        context.startActivity(intent)
                    }
            ) {
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(Dimensions.Padding.content),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Row(
                        horizontalArrangement = Arrangement.spacedBy(Spacing.md),
                        verticalAlignment = Alignment.CenterVertically,
                        modifier = Modifier.weight(1f)
                    ) {
                        Icon(
                            Icons.Default.BugReport,
                            contentDescription = null,
                            tint = MaterialTheme.colorScheme.primary
                        )
                        Column {
                            Text(
                                text = "Report an Issue",
                                style = MaterialTheme.typography.bodyLarge,
                                fontWeight = FontWeight.Medium
                            )
                            Text(
                                text = "Submit bug reports, bank requests, or feature improvements on GitHub",
                                style = MaterialTheme.typography.bodySmall,
                                color = MaterialTheme.colorScheme.onSurfaceVariant
                            )
                        }
                    }
                    Icon(
                        Icons.AutoMirrored.Filled.OpenInNew,
                        contentDescription = null,
                        modifier = Modifier.size(Dimensions.Icon.medium),
                        tint = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
            }
            
            Spacer(modifier = Modifier.height(Spacing.lg))
        }
    }
}

File: iosApp/iosApp/Screens/Settings/FAQScreen.swift
import SwiftUI

struct FAQScreen: View {
    var body: some View {
        List {
            Section {
                FAQCategoryView(
                    icon: "arrow.left.arrow.right",
                    iconColor: .blue,
                    title: "Transaction Types",
                    items: [
                        FAQItem(
                            question: "Why are wallet transactions marked as Credit?",
                            answer: "Wallet transactions (Google Pay, PhonePe, etc.) are marked as Credit because they're charged to your bank account or credit card first, not direct bank debits. The actual bank debit is a separate transaction."
                        ),
                        FAQItem(
                            question: "What's the difference between the 5 transaction types?",
                            answer: "Expense: Money going out (purchases, bills)\nIncome: Money coming in (salary, refunds)\nInvestment: Mutual funds, stocks, SIPs\nCredit: Credit card charges, wallet payments\nTransfer: Moving money between your own accounts"
                        ),
                        FAQItem(
                            question: "When should I use Transfer vs Expense?",
                            answer: "Use Transfer when moving money between your own accounts (e.g., savings to checking). Transfers don't affect your net worth. Use Expense for actual spending where money leaves your possession."
                        ),
                    ]
                )
            }

            Section {
                FAQCategoryView(
                    icon: "lock.shield",
                    iconColor: .green,
                    title: "Privacy & Data",
                    items: [
                        FAQItem(
                            question: "Is my financial data secure?",
                            answer: "Yes, all your data stays on your device. We don't have any servers or cloud storage. Your financial information never leaves your phone."
                        ),
                        FAQItem(
                            question: "Can I backup my data?",
                            answer: "Yes! You can export and import your data using the backup features available in Settings."
                        ),
                        FAQItem(
                            question: "What data does the app access?",
                            answer: "PennyWise only accesses data you manually enter. The app does not access any other app data on your device."
                        ),
                    ]
                )
            }

            Section {
                FAQCategoryView(
                    icon: "building.columns",
                    iconColor: .purple,
                    title: "Account Management",
                    items: [
                        FAQItem(
                            question: "What are manual accounts?",
                            answer: "Manual accounts let you track cash, investments, or any accounts where you update balances yourself. They're great for tracking spending that doesn't go through a bank."
                        ),
                        FAQItem(
                            question: "How do I track multiple accounts?",
                            answer: "You can add separate accounts with different names in Account Management. Each account tracks its own transactions and balance independently."
                        ),
                    ]
                )
            }

            Section {
                FAQCategoryView(
                    icon: "hand.tap",
                    iconColor: .orange,
                    title: "Quick Add",
                    items: [
                        FAQItem(
                            question: "Can I add an expense by tapping the back of my phone?",
                            answer: "Yes. iOS Back Tap can run any shortcut, and PennyWise opens straight to Add Transaction from a link.\n\n1. Open the Shortcuts app and create a new shortcut\n2. Add the \"Open URLs\" action and enter pennywise://add\n3. Name it something like \"Add expense\"\n4. Go to Settings › Accessibility › Touch › Back Tap\n5. Choose Double Tap or Triple Tap, then pick your shortcut\n\nNow a double tap on the back of your phone opens the Add Transaction screen."
                        ),
                        FAQItem(
                            question: "Can I trigger quick add from somewhere other than Back Tap?",
                            answer: "Anything that runs a shortcut can open pennywise://add — put the shortcut on your Home Screen as an icon, in Control Center, or ask Siri for it by name. It works the same way from all of them."
                        ),
                    ]
                )
            }

            Section {
                VStack(spacing: AppSpacing.md) {
                    Text("Still Need Help?")
                        .font(AppTypography.headline)

                    Text("If you couldn't find what you're looking for, feel free to report an issue on GitHub.")
                        .font(AppTypography.caption)
                        .foregroundStyle(.secondary)
                        .multilineTextAlignment(.center)

                    Button {
                        if let url = URL(string: "https://github.com/nicekid1/pennywiseai-tracker/issues/new/choose") {
                            UIApplication.shared.open(url)
                        }
                    } label: {
                        Label("Report an Issue", systemImage: "exclamationmark.bubble")
                            .font(AppTypography.body)
                            .frame(maxWidth: .infinity)
                    }
                    .buttonStyle(.borderedProminent)
                }
                .padding(.vertical, AppSpacing.sm)
                .frame(maxWidth: .infinity)
            }
        }
        .navigationTitle("FAQ")
        .navigationBarTitleDisplayMode(.inline)
    }
}

// MARK: - FAQ Data

struct FAQItem: Identifiable {
    let id = UUID()
    let question: String
    let answer: String
}

// MARK: - FAQ Category View

struct FAQCategoryView: View {
    let icon: String
    let iconColor: Color
    let title: String
    let items: [FAQItem]

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            Label {
                Text(title)
                    .font(AppTypography.headline)
            } icon: {
                Image(systemName: icon)
                    .foregroundStyle(iconColor)
            }
            .padding(.bottom, AppSpacing.sm)

            ForEach(items) { item in
                FAQItemView(item: item)
            }
        }
    }
}

// MARK: - FAQ Item View

struct FAQItemView: View {
    let item: FAQItem
    @State private var isExpanded = false

    var body: some View {
        DisclosureGroup(isExpanded: $isExpanded) {
            Text(item.answer)
                .font(AppTypography.caption)
                .foregroundStyle(.secondary)
                .padding(.top, AppSpacing.xs)
                .padding(.bottom, AppSpacing.sm)
        } label: {
            Text(item.question)
                .font(AppTypography.body)
                .foregroundStyle(.primary)
        }
    }
}
